### PR TITLE
kvserver: trace lease acquisition as a child

### PR DESCRIPTION
**Pull request by andreimatei**
Before this patch, the span for a lease acquisition was not a child of
the triggering request, and thus the trace of the triggering request did
not include it. There's no reason for that, and I've wanted to see the
lease acquisition more than once, so this patch adds the child
relationship.
Lease acquisition is funky because multiple requests can wait on a
an in-flight lease acquisition, and the lease acquisition can outlive
the request that triggered it (if that request is canceled). Still, the
acquisition is tied enough to the request triggering it to justify
making it a child.

Release note: None

### CODE REVIEW

File: pkg/kv/kvserver/replica_range_lease.go [hunk 1]
```diff
@@ -346,8 +347,13 @@ func (p *pendingLeaseRequest) requestLeaseAsync(
 		p.nextLease = roachpb.Lease{}
 	}
 
-	err := p.repl.store.Stopper().RunAsyncTask(
-		ctx, "storage.pendingLeaseRequest: requesting lease", func(ctx context.Context) {
+	err := p.repl.store.Stopper().RunAsyncTaskEx(
+		ctx,
+		stop.TaskOpts{
+			TaskName:  "pendingLeaseRequest: requesting lease",
+			ChildSpan: true,
```
Comment (tbg) [inline pkg/kv/kvserver/replica_range_lease.go:354]: Add a comment, please.

**Review by tbg — APPROVED** [review]

**Review by abarganier — APPROVED** [review]
:lgtm: modulo tbg's comment

> lease acquisition can outlive the request that triggered it

Out of scope here, but longer term is it worth expanding the ability of `TaskOpts` to enable `FollowsFrom` relationships?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/70152#-:-MjjI1gaAWAvyK8IYm3l:b6oazwi)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei)

**Review by andreimatei — COMMENTED** [review]
bors r+

> Out of scope here, but longer term is it worth expanding the ability of TaskOpts to enable FollowsFrom relationships?

When you don't set `TaskOpts.ChildSpan`, you already get a `FollowsFrom` relationship. As it just so happens, I'm expending the options in the area, so watch this space.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/70152#-:-MjokfwlDo-f2PSe0JA4:btp0n3x)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @tbg)

___
*[pkg/kv/kvserver/replica_range_lease.go, line 354 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/70152#-MjVbxR6yOfJD0jP7dCr-r1-354:-MjokdvS6SxRydFrytb9:bb7f2mf) ([raw file](https://github.com/cockroachdb/cockroach/blob/c418c0e8ffe125da7930630cc0df8628e9c617a6/pkg/kv/kvserver/replica_range_lease.go#L354)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

Add a comment, please.
</blockquote></details>

done

**Comment by yuzefovich** [comment]
Misspelling linter is not happy.

bors r-

**Comment by yuzefovich** [comment]
Fixes up the linter.

bors r+

**Pull request merged**